### ISSUE TITLE
missing file / healpixel from faint Galaxia mocks?

### ISSUE THREAD

**Issue by moustakas** [desihub/desitarget#306]
@apcooper The following "faint star" Galaxia mock catalog appears to be missing even though it's nominally within the DESI footprint (as reported by @julienguy) --
`/global/cscratch1/sd/ioannis/desi/mocks/mws/galaxia/alpha/0.0.5_superfaint/healpix/8/5/563/mock_superfaint_allsky_galaxia_desi_b10_cap_south-8-563.fits`

Can you look into this when you get a chance?

**Comment by apcooper** [comment]
I need to generate mocks for that healpixel, I'll post here when it's done. The mocks were made for the desimodel 0.7.0 footprint, I guess something must have changed since then...

**Comment by moustakas** [comment]
@apcooper:  @forero just reported in #307 ([here](https://github.com/desihub/desitarget/pull/307#issuecomment-384413780)) that the following pixel also appears to be missing:
```
0.0.5_superfaint/healpix/8/0/1/mock_superfaint_allsky_galaxia_desi_b10_cap_south-8-1.fits
```
According to @sbailey (in a conversation off-list) the DESI footprint has not changed, so it may be worth drilling a little deeper to try to figure out what happened...

**Comment by forero** [comment]
Now I got the same missing pixel from the main MWS mock. The puzzling thing is that v0.0.5 seemed to provide the exact DESI footprint. This is from the old no-spectra mocks: 

![image](https://user-images.githubusercontent.com/1848618/39270967-3c15b0ce-489d-11e8-99b6-6f8327cf8334.png)

**Comment by apcooper** [comment]
I'll look into this. The two mocks are made in the same way except for the magnitude range, so pixels missing in one should be missing in the other.

@forero:
> The puzzling thing is that v0.0.5 seemed to provide the exact DESI footprint. 

I'm not sure I follow you -- are you just wondering why this didn't show up before? I'm also wondering that.

**Comment by apcooper** [comment]
Can someone clarify how we're deciding these pixels are in the footprint?

With 18.3:
```python
import desimodel.footprint
NSIDE = 8
tiles = desimodel.footprint.load_tiles()
ipix  = desimodel.footprint.tiles2pix(NSIDE,fact=2**16)

missing_pix = [1,563]
for _ in missing_pix:
    if _ in ipix8:
        print('%4d : in footprint'%(_))
    else:
        print('%4d : not in footprint'%(_))
```

```
   1 : not in footprint
 563 : not in footprint
```

`nside = 8` footprint:
![image](https://user-images.githubusercontent.com/9597656/39303043-a7f1375a-494c-11e8-98bc-8211bebd236e.png)

Tiles 1 and 563 in yellow:
![image](https://user-images.githubusercontent.com/9597656/39303073-c0fce2b2-494c-11e8-86a6-494842c72a56.png)

**Comment by forero** [comment]
@apcooper Right, I was wondering why this didn't show up before.

**Comment by geordie666** [comment]
Could this just be related to passing around different values of the resolution `fact`? For example:

```
NSIDE=8
for i in range(8):
    ipix  = desimodel.footprint.tiles2pix(NSIDE,fact=2**i)
    print('fact: {}, pixel 1: {}, pixel 563: {}'.format(2**i,1 in ipix, 563 in ipix))
```
```
fact: 1, pixel 1: True, pixel 563: True
fact: 2, pixel 1: True, pixel 563: True
fact: 4, pixel 1: False, pixel 563: True
fact: 8, pixel 1: False, pixel 563: True
fact: 16, pixel 1: False, pixel 563: True
fact: 32, pixel 1: False, pixel 563: True
fact: 64, pixel 1: False, pixel 563: False
fact: 128, pixel 1: False, pixel 563: False
```

**Comment by apcooper** [comment]
I guess it might be. Should the mocks provide targets to cover the footprint defined by `fact=1` or should code checking the coverage of the mocks use a large value of `fact`?

**Comment by geordie666** [comment]
I think that would be a question for @julienguy (exactly how he's choosing to define the "mock" footprint). Or for @sbailey to make an executive decision.

**Comment by moustakas** [comment]
I don't know if this is related but the minitest notebook (using `tiles2pix`) has some edge issues, as well --
https://github.com/desihub/two_percent_DESI/blob/master/mini/minitest.ipynb
[scroll down to the section titled "Did all pixels make it from sims -> output spectra?"]

It would be great to eliminate to the greatest extent possible all these edge effects.

**Comment by apcooper** [comment]
Just to add that I'm waiting for @julienguy and @sbailey to give input on the question above, before I generate any extra mock healpixels (since it looks like this isn't blocking new quicksurveys now).

**Comment by julienguy** [comment]
Looking at the doc of  tiles2pix, " fact (int, optional): Factor healpy uses to resolve pixel overlaps. When this is large there are fewer false positives at the expense of run time"
So pixels 1 and 563 are not in the footprint, they are selected by mistake when a low fact value is used.
I find it's about the same time to run with fact=4 or 128. So I suggest we simply change the default value of fact to 128 instead of 4 in desimodel.footprint.tiles2pix.

**Comment by sbailey** [comment]
Summarizing my understanding:

With fact=4, pixel 563 is incorrectly included in the footprint but pixel 1 is not.  desitarget.mock.mockmaker calls tiles2pix with the default fact=4, so I don't understand the provenance of how @forero ended up with his run looking for pixel=1.

If @apcooper determined the footprint for his mocks with fact=2**16 like his example, that would explain why he didn't include pixel 563.

At minimum the determination of the mock footprint should use a `fact` less than or equal to the `fact` used by `desitarget.mock.mockmaker` when calling `desimodel.footprint.tiles2pix`.

Proposal:
* change `desimodel.footprint.tiles2pix` to use a more accurate but still fast `fact=128`.
* both `desitarget.mock.mockmaker` and @apcooper should just use that default and not adjust `fact` on their own [insert your own pun about not changing the facts]

If in the short term @apcooper can easily generate pixels 1 and 563 so that we can move on without requiring new desimodel tags, etc. that would also be helpful.

**Comment by apcooper** [comment]
I use `is_point_in_desi` with the default radius to determine which mock stars are in the footprint, so there is no choice of `fact` involved there. 

I've added nside=8 pixels 1 and 563 to the v0.0.5 an 0.0.5_superfaint mocks at NERSC. There might not be complete coverage of these pixels, because I didn't repeat the underlying runs of galaxia; these were already trimmed to a rough version of the footprint plus some small margin that I clip out with `is_point_in_desi`. For these two extra pixels I just skipped that clipping. 

Now I think about it, I could probably have just generated empty tables for these two pixels.
